src/client/src/tscServer.c

```diff
@@ -357,7 +357,7 @@ void checkBrokenQueries(STscObj *pTscObj) {
       pthread_mutex_lock(&pSql->subState.mutex);
       if (pSql->pSubs) {
         // have sub sql
-        for (int i = 0; i < numOfSub; i++) {
+        for (int i = 0; i < pSql->subState.numOfSub; i++) {
```

Comment:
The `pSql->subState.numOfSub` is the size of the array `pSubs`, which are to be synchronized by the mutex.